src/Model/ModelData.php

```diff
@@ -423,9 +370,9 @@ public function escapeTypeForField(string $field): string
      *  - an SSViewer instance
      *
      * @param string|array|SSViewer $template the template to render into
-     * @param ModelData|array|null $customFields fields to customise() the object with before rendering
+     * @param ModelData|array $customFields fields to customise() the object with before rendering
      */
-    public function renderWith($template, ModelData|array|null $customFields = null): DBHTMLText
+    public function renderWith($template, ModelData|array $customFields = []): DBHTMLText
```

Comment:
There was no need for that to ever be null, there's no distinction between "unset value" and "empty"